packages/utilities/src/BaseComponent.ts

```diff
@@ -126,8 +157,59 @@ export class BaseComponent<P, S> extends React.Component<P, S> {
         return this[refName] = ref;
       };
     }
+
     return this.__resolves[refName];
   }
+
+  /**
+   * Updates the componentRef (by calling it with "this" when necessary.)
+   */
+  protected _updateComponentRef(currentProps: IBaseProps<P>, newProps: IBaseProps<P> = {}) {
+    if (this._shouldUpdateComponentRef &&
+      ((!currentProps && newProps.componentRef) ||
+        (currentProps && currentProps.componentRef !== newProps.componentRef))) {
+
+      if (currentProps && currentProps.componentRef) {
+        currentProps.componentRef(null);
+      }
+
+      if (newProps.componentRef) {
+        newProps.componentRef(this);
+      }
+    }
+  }
+  /**
+   * Warns when a deprecated props are being used.
+   *
+   * @protected
+   * @param {IStringMap} deprecationMap The map of deprecations, where key is the prop name and the value is
+   * either null or a replacement prop name.
+   *
+   * @memberOf BaseComponent
+   */
+  protected _warnDeprecations(deprecationMap: IStringMap) {
+    warnDeprecations(this.className, this.props, deprecationMap);
+  }
+
+  /**
+   * Warns when props which are mutually exclusive with each other are both used.
+   *
+   * @protected
+   * @param {IStringMap} mutuallyExclusiveMap The map of mutually exclusive props.
+   *
+   * @memberOf BaseComponent
+   */
+  protected _warnMutuallyExclusive(mutuallyExclusiveMap: IStringMap) {
+    warnMutuallyExclusive(this.className, this.props, mutuallyExclusiveMap);
+  }
+
+  /**
+   * Standard method for rendering null. Useful to wire up onRender methods to a function ref that
+   * never changes.
+   */
+  protected _onRenderNull(): JSX.Element | null {
+    return null;
```

Comment:
Just as an fyi; I use this quite a bit in my other Button PR where i would like to return a function that renders nothing, but don't want to recreate it over and over (to avoid immutability issues)